flash/vision/classification/data.py

```diff
@@ -13,369 +13,392 @@
+        sample, target = sample
+        return self.common_pre_tensor_transform(sample, self.test_transform), target
+
+    def predict_pre_tensor_transform(self, sample: Any) -> Any:
+        if isinstance(sample, torch.Tensor):
+            return sample
+        return self.common_pre_tensor_transform(sample, self.predict_transform)
+
+    def to_tensor_transform(self, sample) -> Any:
+        sample, target = sample
+        return sample if isinstance(sample, torch.Tensor) else self.to_tensor(sample), target
+
+    def predict_to_tensor_transform(self, sample) -> Any:
+        if isinstance(sample, torch.Tensor):
+            return sample
+        return self.to_tensor(sample)
+
+    def common_post_tensor_transform(self, sample: Any, transform) -> Any:
+        return self._apply_transform(sample, transform, "post_tensor_transform")
+
+    def train_post_tensor_transform(self, sample: Any) -> Any:
+        sample, target = sample
+        return self.common_post_tensor_transform(sample, self.train_transform), target
+
+    def val_post_tensor_transform(self, sample: Any) -> Any:
+        sample, target = sample
+        return self.common_post_tensor_transform(sample, self.valid_transform), target
+
+    def test_post_tensor_transform(self, sample: Any) -> Any:
+        sample, target = sample
+        return self.common_post_tensor_transform(sample, self.test_transform), target
+
+    def predict_post_tensor_transform(self, sample: Any) -> Any:
+        return self.common_post_tensor_transform(sample, self.predict_transform)
+
+    def train_per_batch_transform_on_device(self, batch: Tuple) -> Tuple:
+        batch, target = batch
+        return self._apply_transform(batch, self.train_transform, "per_batch_transform_on_device"), target
 
 
 class ImageClassificationData(DataModule):
     """Data module for image classification tasks."""
 
-    @classmethod
-    def from_filepaths(
-        cls,
-        train_filepaths: Union[str, Optional[Sequence[Union[str, pathlib.Path]]]] = None,
-        train_labels: Optional[Sequence] = None,
-        train_transform: Optional[Callable] = _DEFAULT_TRAIN_TRANSFORMS,
-        valid_split: Union[None, float] = None,
-        valid_filepaths: Union[str, Optional[Sequence[Union[str, pathlib.Path]]]] = None,
-        valid_labels: Optional[Sequence] = None,
-        valid_transform: Optional[Callable] = _DEFAULT_VALID_TRANSFORMS,
-        test_filepaths: Union[str, Optional[Sequence[Union[str, pathlib.Path]]]] = None,
-        test_labels: Optional[Sequence] = None,
-        loader: Callable = _pil_loader,
-        batch_size: int = 64,
+    preprocess_cls = ImageClassificationPreprocess
+    image_size = (196, 196)
+
+    def __init__(
+        self,
+        train_ds: Optional[torch.utils.data.Dataset] = None,
+        valid_ds: Optional[torch.utils.data.Dataset] = None,
+        test_ds: Optional[torch.utils.data.Dataset] = None,
+        predict_ds: Optional[torch.utils.data.Dataset] = None,
```

Comment:
```suggestion
        train_ds: Optional[Dataset] = None,
        valid_ds: Optional[Dataset] = None,
        test_ds: Optional[Dataset] = None,
        predict_ds: Optional[Dataset] = None,
```
simplify the typing, the same for `Tensor`